include/utility.hpp

```diff
@@ -215,12 +216,28 @@ template <class CB_CORE> class TParamVar {
 
   explicit TParamVar(CBVar initialValue) { CB_CORE::cloneVar(_v, initialValue); }
 
+  TParamVar(const TParamVar &other) = delete;
+  TParamVar &operator=(const TParamVar &other) = delete;
+
+  TParamVar(TParamVar &&other) : _v(other._v), _cp(other._cp) {
+    other._cp = nullptr;
+    memset(&other._v, 0, sizeof(CBVar));
+  }
+
+  TParamVar &operator=(TParamVar &&other) {
+    _v = other._v;
+    _cp = other._cp;
+    other._cp = nullptr;
+    memset(&other._v, 0, sizeof(CBVar));
+  }
+
   ~TParamVar() {
     cleanup();
     CB_CORE::destroyVar(_v);
   }
 
   void warmup(CBContext *ctx) {
+    cleanup();
```

Comment:
needed in what scenario?
generally warmup is called on cleaned-up chains strictly.

Reply:
might be a red flag/bug

Reply:
I'm using these in places like CBDrawable (The ref-counted instance created by the GFX.Drawable block) to track captured transform/material values.

Turns out I don't need this cleanup though, so I changed it to an assert to make sure it's not accidentally used wrongly.